File: web/src/tiles/clock.ts
import {Tile} from './base';

class ClockTile extends Tile {
    TILE_TYPE = 'CLOCK';
    SAMPLE_CHARS = [
        "🕐", "🕑", "🕒", "🕓", "🕔", "🕕", "🕖", "🕗", "🕘", "🕙", "🕚", "🕛",
        "🕜", "🕝", "🕞", "🕟", "🕠", "🕡", "🕢", "🕣", "🕤", "🕥", "🕦", "🕧"
    ]
    POWER = 255;
    lastTime: number | null;

    constructor(ctx: ExecutionContext, x: number, y: number) {
        super(ctx, x, y);
        this.lastTime = null;
    }

    startTurn() {
        const currentTime = this.ctx.getTime();
        if (!this.lastTime) {
            this.lastTime = currentTime;
            return;
        }
        if (currentTime - this.lastTime >= 1000) {
            for (let dx = 0; dx < 2; dx++) {
                for (let dy = 0; dy < 2; dy++) {
                    if ((dx == 0 && dy == 0) || (dx != 0 && dy != 0)) {
                        continue;
                    }
                    const targetX = this.x + dx;
                    const targetY = this.y + dy;
                    const targetTile = this.ctx.getTile(targetX, targetY);
                    if (targetTile) {
                        targetTile.acceptEnergy(this.POWER, this.x, this.y);
                    }
                }
            }
        }
    }
}

File: lectrick/tiles/clock.py
from .base import Tile


class Clock(Tile):
    TILE_TYPE = "CLOCK"
    SAMPLE_CHARS = [
        "🕐", "🕑", "🕒", "🕓", "🕔", "🕕", "🕖", "🕗", "🕘", "🕙", "🕚", "🕛",
        "🕜", "🕝", "🕞", "🕟", "🕠", "🕡", "🕢", "🕣", "🕤", "🕥", "🕦", "🕧"
    ]
    POWER = 255

    def __init__(self, x, y, tiles):
        super().__init__(x, y, tiles)
        self.last_time = None

    def start_turn(self, execution_data, *args, **kwargs):
        current_time = execution_data.get_time()
        if not self.last_time:
            self.last_time = current_time
            return
        if current_time - self.last_time >= 1:
            for dx in range(-1, 2):
                for dy in range(-1, 2):
                    if (dx == 0 and dy == 0) or (dx != 0 and dy != 0):
                        continue
                    target_x = self.x + dx
                    target_y = self.y + dy
                    try:
                        tile_index = self.tiles.index((target_x, target_y))
                    except ValueError:
                        continue
                    self.tiles[tile_index].accept_energy(255, self.x, self.y)
            self.last_time = current_time
